Shared/Samples/Geocode offline/GeocodeOfflineView.Model.swift

```diff
@@ -0,0 +1,202 @@
+            return CGPoint(x: .zero, y: -image.size.height)
+        }()
+        
+        /// The locator task used to preform the geocode operations, loaded from a local file.
+        private let locatorTask = LocatorTask(name: "SanDiego_StreetAddress", bundle: .main)
+        
+        /// The viewpoint of the map view, initially centered on San Diego, CA, USA.
+        @Published var viewpoint = Viewpoint(
+            center: Point(x: -13_042_250, y: 3_857_970, spatialReference: .webMercator),
+            scale: 2e4
+        )
+        
+        /// The placement of the callout on the map.
+        @Published var calloutPlacement: CalloutPlacement?
+        
+        /// A Boolean value indicating whether the callout placement should be offsetted, e.g. when the map magnifier is showing.
+        @Published var calloutShouldOffset = false
+        
+        /// The text shown in the callout.
+        @Published private(set) var calloutText: String = ""
+        
+        /// A Boolean value indicating whether to show an error alert.
+        @Published var isShowingErrorAlert = false
+        
+        /// The error shown in the error alert.
+        @Published private(set) var error: Error? {
+            didSet { isShowingErrorAlert = error != nil }
+        }
+        
+        init() {
+            graphicsOverlay.addGraphic(markerGraphic)
+        }
+        
+        // MARK: Methods
+        
+        /// Geocodes a given address and adds a marker with the corresponding address at the result's location.
+        /// - Parameter address: The given text address to geocode.
+        func geocodeSearch(address: String) async {
+            guard let locatorTask else { return }
+            
+            // Create geocode parameters.
+            let geocodeParameters = GeocodeParameters()
+            geocodeParameters.addResultAttributeName("Match_addr")
+            geocodeParameters.minScore = 75
+            
+            do {
+                // Perform geocode using the locator task with the text address and parameters.
+                let geocodeResults = try await locatorTask.geocode(
+                    forSearchText: address,
+                    using: geocodeParameters
+                )
+                
+                if let result = geocodeResults.first,
+                   let displayLocation = result.displayLocation {
+                    // If a result is found, place a marker at the result's location.
+                    let resultText = result.attributes["Match_addr"] as? String ?? ""
+                    updateMarker(to: displayLocation, withText: resultText)
+                    
+                    // Zoom to the extent of the result's location.
+                    viewpoint = Viewpoint(boundingGeometry: displayLocation.extent)
+                } else {
+                    // If no result was found, inform the user with an alert.
+                    throw "No results found"
```

Comment:
I checked the .NET sample and [JS example](https://developers.arcgis.com/javascript/latest/tutorials/search-for-an-address/)

.NET sample displays an alert when there is no result. In our case I'm in favor of creating a separate alert only for the no result message, instead of making it part of the error alert.

<img width="332" alt="image" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/9660181/ffa862fe-716a-45f1-beb7-6431cf77a055">

When there is no result, JS display a no result message upon submit button clicked. As this sample is not showcasing the suggestion API, I think it's an overkill to write a UI to do this.